packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -1407,10 +1407,42 @@ class ResizeImage extends ImageProvider<ResizeImageKey> {
   }
 }
 
+/// The strategy for [Image.network] and [NetworkImage] to decide whether to
+/// display images in <img> elements in a platform view instead of fetching
+/// bytes.
+///
+/// See [Image.network] for more explanation on the impact.
+///
+/// This option is only effective on the Web platform. Other platforms always
+/// display network images by fetching bytes.
+enum WebImgElementStrategy {
+  /// Never use <img> elements.
+  never,
+
+  /// Use <img> elements when fetching bytes is not available.
+  ///
+  /// This strategy uses <img> elements if `header` is empty and the fetch
+  /// encounters errors.
+  whenNecessary,
+
+  /// Always use <img> elements as long as `header` is empty.
+  ///
+  /// This strategy still fetches bytes if `header` is not empty, since <img>
+  /// elements do not support headers.
+  always,
```

Comment:
Since it's not guaranteed that an img tag will be used, I'd call this one `prefer` (or `preferPlatformView` for more specificity). If we call this `prefert`, then `whenNecessary` could be called `avoid`. Essentially the full scale of options could be: `never` (fails on cross-origin images), `avoid`, `prefer`, `always` (fails if headers are not empty).

Reply:
I agree with `prefer` or something similar. But I'm not sure about `avoid`. It seems a little ambiguous and could be interpreted as `never`. I prefer `fallback`.

Another option is `whenNecessary` (only use img when necessary) and `whenPossible` (use img whenever possible). But I don't like the extra verbosity.

Reply:
Regardless of the name, I think we should print a warning when `headers` is present but we can't fetch image bytes. Let's make it easier for users to know why their images are not showing up.

Reply:
Agreed, `fallback` sounds better.